service/matching/matchingEngine.go

```diff
@@ -211,17 +211,17 @@ func (e *matchingEngineImpl) removeTaskListManager(id *taskListID) {
 }
 
 // AddDecisionTask either delivers task directly to waiting poller or save it into task list persistence.
-func (e *matchingEngineImpl) AddDecisionTask(ctx context.Context, addRequest *m.AddDecisionTaskRequest) (bool, error) {
-	domainID := addRequest.GetDomainUUID()
-	taskListName := addRequest.TaskList.GetName()
-	taskListKind := common.TaskListKindPtr(addRequest.TaskList.GetKind())
+func (e *matchingEngineImpl) AddDecisionTask(hCtx *handlerContext, request *m.AddDecisionTaskRequest) (bool, error) {
```

Comment:
Seems like only the regular context is used in these method. Do you foresee that handlerContext can be used in future, maybe for us to add more metrics inside engine?

Reply:
I assume you are talking about the taskListManager.AddTask() calls ? The reason these don't take handlerContext is because the lifetime of taskListManager is beyond the lifetime of request. For this reason, they don't use the metric scope in the handlerContext. They have their own scope (which is per_task_list )